### PR TITLE
added sign-in and signup functions in auth routes and controllers

### LINKED ISSUE #20: login/sign-up and logout APIs
We are introducing a login/sign-up functionality for users using passport and jwt-authentication. This feature will introduce many other possibilities we can provide to respective users. Follow the REST architectural style for implementing these APIs.

### PR DESCRIPTION
**Pull request by majjikishore007**
Fixes #20

### CODE REVIEW

File: controllers/url.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const client = redis.createClient(REDISPORT);
 
 const baseURL = config.get('baseURL');
 
-async function shorten(req, res) {
+ exports.URLshorten= async (req, res)=> {
```
Comment (arjungarg07) [inline controllers/url.js]: Revert these changes and follow only the convention used in the project.

File: middlewares/cachePolicy.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const redis = require('redis');
 const REDISPORT = process.env.REDISPORT || 6379;
 const client = redis.createClient(REDISPORT);
 
-function cache(req,res,next){
+ cache=(req,res,next)=>{
```
Comment (arjungarg07) [inline middlewares/cachePolicy.js]: Revert these changes as these are changes being made outside the domain of the issue you are working on.

File: models/Url.js [hunk 3]
```diff
@@ -1,7 +1,5 @@
 const mongoose = require('mongoose');
-const { Schema } = mongoose;
-
-const urlSchema = new Schema({
+const urlSchema = new mongoose.Schema({
```
Comment (arjungarg07) [inline models/Url.js]: Please follow the convention and have it destructured.

File: routes/auth.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+const router=express.Router();
+
+const { signup, signin, signout }=require("../controllers/auth")
+
+router.post("/signup",signup);
+router.post("signin",signin);
+router.get("signout",signout);
+
+module.exports=router;
```
Comment (arjungarg07) [inline routes/auth.js]: Don't make a new auth file for these routes and add them in index.js itself

**Review by arjungarg07 — CHANGES_REQUESTED** [review]

**Comment by arjungarg07** [comment]
Also, provide the changes you made in the description while making a PR.

File: config/db.js [hunk 5]
```diff
@@ -4,16 +4,16 @@ const db = config.get('mongoURI');
...
-	}
-  };
+		 useUnifiedTopology: true,
+		 useCreateIndex:true
+		}).then(()=>{
+			console.log("DB IS CONNECTED sucessfully");
+		}).catch(()=>{
+			console.log("DB connection failed");
+		})
+}
```
Comment (arjungarg07) [inline config/db.js]: revert these changes, we are using await so don't use .then() with it.No changes required in this file.

**Comment by BalgopalPatro** [comment]
I think we are going to use passport.js for authentication ? @arjungarg07

File: controllers/url.js (hunk 1 above)
Comment (majjikishore007) [reply controllers/url.js]: revetred the changes

File: routes/auth.js (hunk 4 above)
Comment (majjikishore007) [reply routes/auth.js]: added the routes in index.js

**Comment by arjungarg07** [comment]
Yes @BalgopalPatro

File: index.html [hunk 6]
```diff
@@ -1,32 +1,43 @@
...
-		  <span id="invalid-url" class="text-white text-left m-6 px-10"></span>
-		  <a class="text-white text-center m-6 px-10" href="#" id="shorten_url" target="_blank"></a>
-		  <button class="hidden text-white bg-indigo-500 border-0 py-2 px-10 m-6 focus:outline-none hover:bg-indigo-600 rounded text-lg" id="copy">Copy</button>
-	  	</div>
-    </section>
-    <script src="/js/index.js"></script>
-    
-  </body>
-</html>
+</head>
```
Comment (arjungarg07) [inline index.html:10]: Revert these changes, indentation fix will be done afterwords

**Comment by majjikishore007** [comment]
resolved confilcts

File: index.html (hunk 6 above)
Comment (arjungarg07) [reply index.html:10]: @majjikishore007 revert these changes.

**Review by arjungarg07 — CHANGES_REQUESTED** [review]
